spec/features/portal/client_portal_dashboard_spec.rb

```diff
@@ -1,44 +1,69 @@
 require "rails_helper"
 
 RSpec.feature "a client on their portal" do
-  let(:client) { create :client, intake: (create :intake, preferred_name: "Martha", primary_first_name: "Martha", primary_last_name: "Mango", filing_joint: "yes") }
-  let(:tax_return2019) { create :tax_return, :ready_to_sign, year: 2019, client: client }
-  let(:tax_return2018) { create :tax_return, :ready_to_file_solo, year: 2018, client: client }
-  before do
-    create :document, display_name: "Another 8879", document_type: DocumentTypes::UnsignedForm8879.key, tax_return: tax_return2019, client: tax_return2019.client
-    create :tax_return, year: 2017, client: client
-    create :document, document_type: DocumentTypes::FinalTaxDocument.key, tax_return: tax_return2019, client: tax_return2019.client
-    create :document, document_type: DocumentTypes::FinalTaxDocument.key, display_name: "Some final tax document", tax_return: tax_return2018, client: tax_return2018.client
-    create :document, document_type: DocumentTypes::FinalTaxDocument.key, display_name: "Another final tax document", tax_return: tax_return2018, client: tax_return2018.client
-    login_as client, scope: :client
+  context "when a client has not yet completed onboarding and next step is a question" do
+    let(:client) { create :client, intake: (create :intake, preferred_name: "Katie", current_step: "/en/questions/asset-loss") }
+    before do
+      login_as client, scope: :client
+    end
+    scenario "linking to next step" do
+      visit portal_root_path
+      expect(page).to have_text "Welcome back Katie!"
+      expect(page).to have_link("Complete all tax questions", href: "/en/questions/asset-loss")
+    end
   end
 
-  scenario "viewing their tax return statuses" do
-    visit portal_root_path
-    expect(page).to have_text "Welcome back Martha!"
+  context "when a client has not yet completed onboarding and next step is documents" do
+    let(:client) { create :client, intake: (create :intake, preferred_name: "Randall", current_step: "/en/documents/overview") }
+    before do
+      login_as client, scope: :client
+    end
+    scenario "linking to next step" do
+      visit portal_root_path
+      expect(page).to have_text "Welcome back Randall!"
+      expect(page).to have_link("Submit remaining tax documents", href: "/en/documents/overview" )
+    end
+  end
```

Comment:
I'd recommend moving these feature specs into the portal controller spec with a render_views, as they are mainly testing the template conditional statement.

I think there is a bigger, more important feature spec to add later, in which a client partially fills out intake, drops off, returns and signs into the portal, then sees one of these links and follows it to complete their intake, but that can be added with a later story.

Reply:
I wrote these as separate feature specs because to my mind they are two distinct client flows that will continue to diverge more and more as we build out these features. Also, I think render_views in controller tests should be used sparingly, as it's kind of seen as a "last resort" feature in rails -- thats why Rails makes you load it in as an option instead of being on by default.